completions/xrandr

```diff
@@ -1,62 +1,165 @@
+}
+
+_xrandr_providers()
+{
+    "$1" --listproviders 2>/dev/null | sed -ne "s/.*cap:.*$2.* name:\([^ ]*\).*/\1/p"
+}
+
+_xrandr_modes()
+{
+    "$1" -q 2>/dev/null | sed -e "1,/^$2 / d" \
+                              -e '/connected/,$ d' \
+                              -e '/^[[:space:]]*h: / d' \
+                              -e '/^[[:space:]]*v: / d' \
+                              -e 's/\([^[:space:]]\)[[:space:]].*/\1/'
+}
+
+_xrandr_help()
+{
+    "$1" --help 2>/dev/null
+}
+
 _xrandr()
 {
     local cur prev words cword
     _init_completion || return
 
+    local i output
+    for ((i = cword - 1; i > 0; i--)); do
+        if [[ ${words[i]} == --output ]]; then
+            output=${words[i + 1]}
+            break
+        fi
+    done
+
     case "$prev" in
-        -display | -d | -help | -s | --size | -r | --rate | --refresh | --screen | --fb | --fbmm | \
-            --dpi | --pos | --set | --scale | --transform | --crtc | --panning | --gamma | \
-            --newmode | --rmmode | --addmode | --delmode)
+        -display | -d | -s | --size | -r | --rate | --refresh | --screen | --fb | --fbmm | \
+            --pos | --set | --scale | --transform | --crtc | --panning | --gamma | \
+            --newmode | --rmmode | --setmonitor)
+            return
+            ;;
+        --output | --addmode | --delmode)
+            local outputs=$(_xrandr_outputs "$1")
+            COMPREPLY=($(compgen -W "$outputs" -- "$cur"))
             return
             ;;
-        --output | --left-of | --right-of | --above | --below | --same-as)
-            local outputs=$("$1" | awk '/connected/ {print $1}')
+        --dpi)
+            local outputs=$(_xrandr_outputs "$1")
             COMPREPLY=($(compgen -W "$outputs" -- "$cur"))
             return
             ;;
+        --left-of | --right-of | --above | --below | --same-as)
+            if [[ -v output ]]; then
+                local outputs=$(_xrandr_outputs "$1")
+                COMPREPLY=($(compgen -W "$outputs" -- "$cur"))
+            fi
+            return
+            ;;
         --mode)
-            local i output
-            for ((i = 1; i < cword; i++)); do
-                if [[ ${words[i]} == --output ]]; then
-                    output=${words[i + 1]}
-                    break
-                fi
-            done
             if [[ -v output ]]; then
-                local modes=$("$1" | command sed -e "1,/^$output / d" \
-                    -e "/connected/,$ d" \
-                    -e "s/\([^[:space:]]\)[[:space:]].*/\1/")
+                local modes=$(_xrandr_modes "$1" "$output")
                 COMPREPLY=($(compgen -W "$modes" -- "$cur"))
             fi
             return
             ;;
         -o | --orientation)
-            COMPREPLY=($(compgen -W 'normal inverted left right 0 1 2 3' -- \
-                "$cur"))
+            COMPREPLY=($(compgen -W 'normal inverted left right 0 1 2 3' -- "$cur"))
```

Comment:
This seems like an unrelated whitespace change, in the future please keep them as separate commits.

Reply:
In general it's a good thing to do one thing in one commit, here everything's lumped in one. Just something to remember for future changes.

Reply:
Sorry, I don't know why I removed that newline. I can put it back if necessary.

Reply:
No need, as said, just for the future.